Get Hello world page ready for section tagging and filter (#7386)
circleci/circleci-docs: jekyll/_cci2/hello-world.adoc

---
redirect_from: /examples-intro/
tags:
  platform:
    - Cloud
    - Server v4.x
    - Server v3.x
    - Server v2.x
sectionTags:
  Cloud:
    - "#echo-hello-world-cloud"

>   Server_v4:
>     - "#echo-hello-world-server-4"
>   Server_v3:
>     - "#echo-hello-world-server-3"
>   Server_v2:

Author: ```suggestion
  Server v4.x:
    - "#echo-hello-world-server-4"
  Server v3.x:
    - "#echo-hello-world-server-3"
  Server v2.x:
```

    - "#echo-hello-world-server-2"
---
= Hello world
:page-layout: classic-docs
:page-liquid:
:page-description: Get started with CircleCI by printing Hello World in any execution environment.
:icons: font

> This page describes how to get started with a basic pipeline using any execution environment. The steps described here assume you want to create a new project locally and push your changes to your VCS using `git`. This page assumes you have already link:/docs/first-steps/[signed up] for CircleCI.

Reviewer: ```suggestion
This page shows you how to get started with a basic pipeline using any execution environment. The steps described here assume you want to create a new project locally and push your changes to your VCS using Git. This page also assumes you have already link:/docs/first-steps/[signed up] for CircleCI.
```
Nitpicks: two instances of "describe" right by each other 🙂 Also we probably don't need to use special formatting with Git

---
redirect_from: /examples-intro/
tags:
  platform:
    - Cloud
    - Server v4.x
    - Server v3.x
    - Server v2.x
sectionTags:
  Cloud:
    - "#echo-hello-world-cloud"
  Server_v4:
    - "#echo-hello-world-server-4"
  Server_v3:
    - "#echo-hello-world-server-3"
  Server_v2:
    - "#echo-hello-world-server-2"
---
= Hello world
:page-layout: classic-docs
:page-liquid:
:page-description: Get started with CircleCI by printing Hello World in any execution environment.
:icons: font

This page describes how to get started with a basic pipeline using any execution environment. The steps described here assume you want to create a new project locally and push your changes to your VCS using `git`. 

> As an alternative, if you already have a repository you want to "plug in" to CircleCI you can set this up entirely within the CircleCI web app, including commiting a config file. This flow is outlined in the link:/docs/getting-started[Quickstart guide].

Reviewer: I think this paragraph could be moved to after the next one (i.e. "This page assumes you have already signed up for CIrcleCI)

This page assumes you have already link:/docs/first-steps/[signed up] for CircleCI.

[#echo-hello-world-cloud]
== Echo hello world

This example adds a job called `hello-job` that simply prints `hello world` to the console. To get started, complete the following steps:

. Create a directory called `.circleci` in the root directory of your local code repository.

. From the `.circleci` directory you just created, create a link:/docs/configuration-reference[`config.yml`] file and add the following content (select your execution environment using the tabs):
+
[.tab.hellocloud.Docker]
--

> The job `hello-job` spins up a container running a link:/docs/circleci-images/#nodejs[pre-built CircleCI Docker image for Node]. Refer to link:/docs/using-docker[Using the Docker execution environment] page for more information.

Reviewer: Is there a way to leave a bit more spacing between the tabs and the text within each tab? The GPU one looks fine to me but all the others look a bit squished.

The text wrapping in the tabs also looks a bit off. When I have my browser at a certain smaller size, the sentence "The job `hello-job`..." doesn't include the period in the wrapping so it shows up in the next line, separated from the sentence.

Author: Yes the spacing between tab and text is something I've asked DD about before, I've just asked again. I'll ask about wrapping too :-D

[source,yaml]
----
version: 2.1

jobs:
  hello-job:
    machine:
      image: ubuntu-2004-cuda-11.4:202110-01
    steps:
      - checkout # check out the code in the project directory
      - run: echo "hello world" # run the `echo` command

workflows:
  my-workflow:
    jobs:
      - hello-job
----
--
+
[.tab.hellocloud.Arm]
--
The job `hello-job` spins up an [Arm (Linux) virtual machine] using the machine executor. Refer to link:/docs/using-arm[Using the Arm execution environment] page for more information.

[source,yaml]
----
version: 2.1

jobs:
  hello-job:
    machine:
      image: ubuntu-2004:202101-01
    resource_class: arm.medium
    steps:
      - checkout # check out the code in the project directory
      - run: echo "hello world" # run the `echo` command

workflows:
  my-workflow:
    jobs:
      - hello-job
----
--

. Commit and push the changes to your VCS.

> . Go to the **Projects** page in the CircleCI web app, then click the **Set Up Project** button next to your new project. If you do not see your project, make sure you have selected the correct organization at the top-left of the web app. See the link:/docs/introduction-to-the-circleci-web-app#organizaiton-switching[Organization Switching] for more information.

Reviewer: ```suggestion
. Go to the **Projects** page in the CircleCI web app, then click the **Set Up Project** button next to your new project. If you do not see your project, make sure you have selected the correct organization at the top-left of the web app. See the link:/docs/introduction-to-the-circleci-web-app#organization-switching[Organization Switching] for more information.
```
"Organization Switching page"? Also do we stick to title case or sentence case?

Author: I'll stick with title case for now but if/when we decide to change we can hopefully work out a way to make a global change

. Follow the steps in the pop-up to tell CircleCI to use the `config.yml` file you just created to trigger your first pipeline. Clicking through to the `hello-job` and then selecting the `echo "hello world" step` will show you `hello world` in the console.
+
image:hello-world-app.png[Screenshot showing hello world in the job step output]

**Tip:** If you get a `No Config Found` error, it may be that you used `.yaml` file extension. Be sure to use `.yml` file extension to resolve this error.

[#echo-hello-world-server-4]

> == Echo hello world (server v4.x)

Reviewer: Do we also need the server version in the section heading? (I am assuming it's because we can't have duplicate heading values within the same document, even if the anchors/ids are unique ..?)

Author: I had left it in because I am not yet sure what the default state would be when tagging is implemented (e.g. default all visible until you select an option, or default just Cloud is visible), and these changes will go in pre-tagging so people need to be able to see what each section is for.

Reviewer: Gotcha! Not sure how I feel about parentheses in the heading, do you think it would be better to remove them and say "on"/"in" instead..?

Author: Yeah good idea, will do!

      - hello-job
----
--
+
[.tab.helloserver4.Windows]
--
The job `hello-job` spins up a Windows virtual machine using the default executor specified by the link:https://circleci.com/developer/orbs/orb/circleci/windows#usage-run_default[Windows orb]. Refer to link:/docs/using-windows[Using the Windows execution environment] page for more information.

[source,yaml]
----
version: 2.1

jobs:
  hello-job:
    machine:
      image: windows-default

    steps:
      # Commands are run in a Windows virtual machine environment
      - checkout
      - run: Write-Host 'Hello, Windows'

workflows:
  my-workflow:
    jobs:
      - hello-job
----
--
+
[.tab.helloserver4.Arm]
--
The job `hello-job` spins up an Arm (Ubuntu 22.04) virtual machine. Refer to link:/docs/using-arm[Using the Arm execution environment] page for more information.

[source,yaml]
----
version: 2.1

jobs:
  hello-job:
    machine:
      image: arm-default
    resource_class: arm.medium
    steps:
      - checkout # check out the code in the project directory
      - run: echo "hello world" # run the `echo` command

workflows:
  my-workflow:
    jobs:
      - hello-job
----
--

. Commit and push the changes to your VCS.

> . Go to the **Projects** page in the CircleCI web app, then click the **Set Up Project** button next to your new project. If you do not see your project, make sure you have selected the correct organization at the top-left of the web app. See the link:/docs/introduction-to-the-circleci-web-app#organizaiton-switching[Organization Switching] for more information.

Reviewer: ```suggestion
. Go to the **Projects** page in the CircleCI web app, then click the **Set Up Project** button next to your new project. If you do not see your project, make sure you have selected the correct organization at the top-left of the web app. See the link:/docs/introduction-to-the-circleci-web-app#organization-switching[Organization Switching] for more information.
```

      - hello-job
----
--
+
[.tab.helloserver3.Windows]
--
The job `hello-job` spins up a Windows virtual machine using the default executor specified by the link:https://circleci.com/developer/orbs/orb/circleci/windows#usage-run_default[Windows orb]. Refer to link:/docs/using-windows[Using the Windows execution environment] page for more information.

[source,yaml]
----
version: 2.1

jobs:
  hello-job:
    machine:
      image: windows-default

    steps:
      # Commands are run in a Windows virtual machine environment
      - checkout
      - run: Write-Host 'Hello, Windows'

workflows:
  my-workflow:
    jobs:
      - hello-job
----
--
+
[.tab.helloserver3.Arm]
--
The job `hello-job` spins up an Arm (Ubuntu 22.04) virtual machine. Refer to link:/docs/using-arm[Using the Arm execution environment] page for more information.

[source,yaml]
----
version: 2.1

jobs:
  hello-job:
    machine:
      image: arm-default
    resource_class: arm.medium
    steps:
      - checkout # check out the code in the project directory
      - run: echo "hello world" # run the `echo` command

workflows:
  my-workflow:
    jobs:
      - hello-job
----
--

. Commit and push the changes to your VCS.

> . Go to the **Projects** page in the CircleCI web app, then click the **Set Up Project** button next to your new project. If you do not see your project, make sure you have selected the correct organization at the top-left of the web app. See the link:/docs/introduction-to-the-circleci-web-app#organizaiton-switching[Organization Switching] for more information.

Reviewer: ```suggestion
. Go to the **Projects** page in the CircleCI web app, then click the **Set Up Project** button next to your new project. If you do not see your project, make sure you have selected the correct organization at the top-left of the web app. See the link:/docs/introduction-to-the-circleci-web-app#organization-switching[Organization Switching] for more information.
```

--
The job `hello-job` spins up a linux vir†ual machine running a link:https://circleci.com/developer/images?imageType=machine[Ubuntu machine image]. Refer to link:/docs/using-linuxvm[Using the Linux VM execution environment] page for more information.

[source,yaml]
----
version: 2.1

jobs:
  hello-job:
    machine: true
    steps:
      - checkout # check out the code in the project directory
      - run: echo "hello world" # run the `echo` command

workflows:
  my-workflow:
    jobs:
      - hello-job
----
--
+
[.tab.helloserver2.Windows]
--
The job `hello-job` spins up a Windows virtual machine using the default executor specified by the link:https://circleci.com/developer/orbs/orb/circleci/windows#usage-run_default[Windows orb]. Refer to link:/docs/using-windows[Using the Windows execution environment] page for more information.

[source,yaml]
----
version: 2.1

jobs:
  hello-job:
    machine:
      image: windows-default

    steps:
      # Commands are run in a Windows virtual machine environment
      - checkout
      - run: Write-Host 'Hello, Windows'

workflows:
  my-workflow:
    jobs:
      - hello-job
----
--

. Commit and push the changes to your VCS.

> . Go to the **Add Projects** page in the CircleCI web app, then click the **Start Building** button. If you do not see your project, make sure you have selected the correct organization at the top-left of the web app. See the link:/docs/introduction-to-the-circleci-web-app#organizaiton-switching[Organization Switching] for more information. You will then see your job and workflow in the relevant pages in the app

Reviewer: ```suggestion
. Go to the **Add Projects** page in the CircleCI web app, then click the **Start Building** button. If you do not see your project, make sure you have selected the correct organization at the top-left of the web app. See the link:/docs/introduction-to-the-circleci-web-app#organization-switching[Organization Switching] for more information. You will then see your job and workflow in the relevant pages in the app
```
